### PR TITLE
don't use the ODE solver for the dae_init.

### PR DESCRIPTION
**Pull request by oscardssmith**
Not sure how this slipped through.

### CODE REVIEW

**Review by YingboMa — CHANGES_REQUESTED** [review]
format

**Comment by ChrisRackauckas** [comment]
And we should find a way to test this.

**Comment by oscardssmith** [comment]
any testing suggestions? the only thing I can think of is making a very broken nonlinear solver in the tests.

**Comment by oscardssmith** [comment]
test added.

File: src/initialize_dae.jl [hunk 1]
```diff
@@ -211,37 +211,19 @@ function _initialize_dae!(integrator, prob::ODEProblem, alg::ShampineCollocation
...
-        @.. broadcast=false integrator.u=integrator.uprev + z
-    else
-        nlequation_oop = @closure (u, _) -> begin
-            update_coefficients!(M, u, p, t)
-            M * (u - u0) / dt - f(u, p, t)
-        end
+    nlequation_oop = @closure (u, _) -> begin
+        update_coefficients!(M, u, p, t)
+        M * (u - u0) / dt - f(u, p, t)
+    end
```
Comment (ChrisRackauckas) [inline src/initialize_dae.jl:217]: This isn't right, it should default to using the nonlinear solver.

Comment (oscardssmith) [reply src/initialize_dae.jl:217]: isn't that what `nlsolve = default_nlsolve(alg.nlsolve, isinplace, u0)` does?

Comment (ChrisRackauckas) [reply src/initialize_dae.jl:217]: No, my point is that you shouldn't use NonlinearSolve.jl unless you have to. It should just default to using the internal one.

Comment (oscardssmith) [reply src/initialize_dae.jl:217]: ok, so the order you want is `initializealg.nlsolve > integrator.cache.nlsolver > default`. Should this change also be made to the rest of the dae initialization methods?

Comment (oscardssmith) [reply src/initialize_dae.jl:217]: wait, even if the user passes a nlsolve to their initialization, you want us to ignore that if the integrator has a solver?

Comment (ChrisRackauckas) [reply src/initialize_dae.jl:217]: yes, that's the point of passing something that overrides the default right?

**Review by ChrisRackauckas — APPROVED** [review]

**Pull request merged**